app/assets/javascripts/discourse/components/date-field.js.es6

```diff
@@ -0,0 +1,4 @@
+
+export default Ember.TextField.extend({
+    type: 'date'
```

Comment:
If all you're doing is changing the type of field, you can do it in the hbs template instead. We don't need the `date-field` and `time-field` components.

`{{input type="date" value=something}}`